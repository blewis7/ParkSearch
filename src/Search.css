.header {
    width: 100vw;
    height: 40vh;
    background-size: 100% 100%;
    padding: 0;
    margin: 0;
    background-color: black;
}

.search {
    margin-top: -10vh;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100vw;
    height: 50vh;
    flex-direction: column;
}

h1 {
    flex-direction: column;
    color: rgb(138, 91, 30);
    font-size: 2.6rem;
}

.launch {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: row;
}

input {
    width: 40vw;
}

.search-button{
    background-color: forestgreen;
    color: white;
}


.suggestions {
    display: flex;
    position: absolute;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    z-index: 1;
    background-color: white;
}

.suggestion {
    border-style: outset;
    width: 39.9vw;
}

.suggestion:hover {
    cursor: pointer;
    background-color: lightgrey;
}

.search-results {
    display: flex;
    flex-direction: column;
    width: 100vw;
    border: black;
    align-items: center;
    justify-content: center;
    z-index: 2;
}

.results:hover {
    text-decoration: none;
}

.search-results a {
    display: flex;
    text-decoration: none;
    width: 400px;
    color: black;
    margin-bottom: 40px;
    padding: 20px;
    text-align: center;
    border: 2px solid rgba(0, 0, 0, 0.237);
    border-radius: 15%;

}

.search-results a:hover {
    background-color: lightblue;
    text-decoration: none;
    color: black;
}