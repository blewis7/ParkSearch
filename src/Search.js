import React, {useState, useEffect} from "react";
import axios from "axios";
import { ApiKey } from "./ApiKey";
import "./Search.css";
import forest from "./forest.png"

function Search(){
    const [parks, setParks] = useState([]);
    const [term, setTerm] = useState("");
    const [suggestions, setSuggestions] = useState([]);
    const [results, setResults] = useState([]);

    useEffect(() => {
        const loadParks = async () => {
            const response = await axios.get(`https://developer.nps.gov/api/v1/parks?limit=500&${ApiKey}`);
            setParks(response.data.data);
        }
        loadParks();
    }, [])

    const handleChange = (term) => {
        let matches = [];
        if (term.length > 0){
            matches = parks.filter(park => {
                const regex = new RegExp(`${term}`, "gi");
                return park.fullName.match(regex);
            })
        }
        if (matches.length > 50){
            setSuggestions([]);
        } else {
            setSuggestions(matches);
        }
        setTerm(term);
    }

    const updateSearchTerm = (term) => {
        setTerm(term);
        setSuggestions([]);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        let matches = [];
        if (term.length > 0 && term !== " "){
            matches = parks.filter(park => {
                const regex = new RegExp(`${term}`, "gi");
                return (park.fullName.match(regex) || park.states.match(regex));
            })
        }
        setResults(matches);
        setSuggestions([]);
        setTerm("");
    }


    return (
        <>
            <div className="header" style={{backgroundImage: `url(${forest})`}}/>
            <div className="search">
                <h1><b>Search National Parks</b></h1>
                <form onSubmit={handleSubmit}>
                    <div className="launch mt-5">
                        <div className="autocomplete">
                            <input type="text" 
                                placeholder="Search by State Code (ex: CA) or Park Name"
                                onChange={e => handleChange(e.target.value)}
                                value={term}
                                onBlur={() => {
                                    setTimeout(() => {
                                    setSuggestions([])
                                    }, 250);
                                }}
                            />
                            <div className="suggestions">
                                {suggestions && suggestions.map((suggestion, i) =>
                                    <div className="suggestion" key={i}
                                        onClick={() => updateSearchTerm(suggestion.fullName)}
                                    >{suggestion.fullName}</div>
                                )}
                            </div>
                        </div>
                        <button className="search-button">Search</button>
                    </div>
                </form>
            </div>
            <div className="search-results">
                {results && results.map((result) =>
                    <a href={`/${result.parkCode}`}  className="result" key={result.parkCode}>
                        <div>
                            <h2>{result.fullName}</h2>
                            <p>{result.description}</p>
                            <h4>States: {result.states.substring(0, 23)}</h4>
                        </div>
                    </a>
                )}
            </div>
        </>
    )
}

export default Search;
