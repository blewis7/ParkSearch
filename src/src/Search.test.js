import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search";

// smoke test
it("renders without crashing", function() {
  render(<Search />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<Search />);
  expect(asFragment()).toMatchSnapshot();
});

it("will show appropriate elements", function() {
    const search = render(<Search />);

    expect(search.queryByText("Search")).toBeInTheDocument();
    expect(search.getByTestId("my-input")).toBeInTheDocument();
});

