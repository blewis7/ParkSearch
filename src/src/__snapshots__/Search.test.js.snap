// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
    style="background-image: url(forest.png);"
  />
  <div
    class="search"
  >
    <h1>
      <b>
        Search National Parks
      </b>
    </h1>
    <form>
      <div
        class="launch mt-5"
      >
        <div
          class="autocomplete"
        >
          <input
            data-testid="my-input"
            placeholder="Search by State Code (ex: CA) or Park Name"
            type="text"
            value=""
          />
          <div
            class="suggestions"
          />
        </div>
        <button
          class="search-button"
        >
          Search
        </button>
      </div>
    </form>
  </div>
  <div
    class="search-results"
  />
</DocumentFragment>
`;
